src/coreclr/jit/block.h

```diff
@@ -702,8 +702,26 @@ struct BasicBlock : private LIR::Range
     // a block corresponding to an exit from the try of a try/finally.
     bool isBBCallAlwaysPairTail() const;
 
+private:
     BBjumpKinds bbJumpKind; // jump (if any) at the end of this block
 
+public:
+    BBjumpKinds getBBJumpKind() const
+    {
+        return bbJumpKind;
+    }
+
+    void setBBJumpKind(BBjumpKinds kind DEBUG_ARG(Compiler* comp))
+    {
+#ifdef DEBUG
```

Comment:
I'm probably getting ahead of myself here, but I'm thinking we will need to track whether BBJ_NONE is allowed or not in the current Compiler's state, so passing in a pointer to it will enable us to assert this state. I avoided making this a noway_assert to minimize TP diffs, and made `comp` a debug arg to avoid unused variable warnings in Release builds (though this approach also has the negative effect of removing this assert in Checked builds).

Reply:
I would personally recommend using a verifier pattern (i. e. adding an assert somewhere in fgdiagnostic.cpp) for checking this property instead of the setters. It will make for less debug-only code overall.

Reply:
I like the cleanliness of your suggestion, but if we move this check to somewhere like `Compiler::fgDebugCheckBBlist` instead of checking every time `bbJumpKind` is updated, then is there any need to make `bbJumpKind` private? @AndyAyersMS

Reply:
> is there any need to make bbJumpKind private

Not strictly, but it is still an improvement.

https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/clr-jit-coding-conventions.md

> 15.5.1 Public data members
> Do not declare public data members. Instead, public accessor functions should be exposed to access class members.